plugins/rest/aws.go

```diff
@@ -451,17 +485,17 @@ func signV4(req *http.Request, credService awsCredentialService, theTime time.Ti
 
 	// the "string to sign" is a time-bounded, scoped request token which
 	// is linked to the "canonical request" by inclusion of its SHA-256 hash
-	strToSign := "AWS4-HMAC-SHA256\n"                                    // V4 signing with SHA-256 HMAC
-	strToSign += iso8601Now + "\n"                                       // ISO 8601 time
-	strToSign += dateNow + "/" + creds.RegionName + "/s3/aws4_request\n" // scoping for signature
-	strToSign += fmt.Sprintf("%x", sha256.Sum256([]byte(canonicalReq)))  // SHA-256 of canonical request
+	strToSign := "AWS4-HMAC-SHA256\n"                                                 // V4 signing with SHA-256 HMAC
+	strToSign += iso8601Now + "\n"                                                    // ISO 8601 time
+	strToSign += dateNow + "/" + creds.RegionName + "/" + service + "/aws4_request\n" // scoping for signature
+	strToSign += fmt.Sprintf("%x", sha256.Sum256([]byte(canonicalReq)))               // SHA-256 of canonical request
```

Comment:
Hmm this is getting a bit unwieldy... I guess alternatives would be _one giant sprintf_, or using a `strings.Builder` 🤔

Reply:
I had a quick look at different options for string formatting in Go and while a little ugly I feel like string concatenation is probably the most intuitive looking way to write it... sprintf moves the variables to the end which adds a little overhead for the reader, and while a string builder is more efficient, it's quite a lot more verbose 😛

Reply:
I'll take your word for it, thanks for checking 👍